File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
alphabet_list = ['A', 'E', 'I', 'O', 'U']
idx, result = 0, 0

def dfs(cnt, alphabet, word):
    global idx, result
    
    if alphabet == word:
        result = idx
        return
    
    if cnt == 5:
        return
    
    for i in range(5):
        idx += 1
        dfs(cnt + 1, alphabet + alphabet_list[i], word)

def solution(word):
    global result
    
    dfs(0, '', word)
    return result

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    
    private char[] alphabetList = new char[]{'A', 'E', 'I', 'O', 'U'};
    private int idx = 0;
    private int result = 0;
    
    public void dfs(int cnt, String alphabet, String word) {
        
        if (alphabet.equals(word)) {
            result = idx;
            return;
        }
        
        if (cnt == 5) {
            return;
        }
        
        for (int i = 0; i < 5; i++) {
            idx++;
            dfs(cnt + 1, alphabet + alphabetList[i], word);
        }
    }
    
    public int solution(String word) {
        dfs(0, "", word);
        return result;
    }
}